graphrag/general/extractor.py

```diff
@@ -53,19 +55,52 @@ def __init__(
         self._get_relation_ = get_relation
         self._set_relation_ = set_relation
 
-    def _chat(self, system, history, gen_conf):
+    async def _chat(self, system, history, gen_conf):
         hist = deepcopy(history)
         conf = deepcopy(gen_conf)
         response = get_llm_cache(self._llm.llm_name, system, hist, conf)
         if response:
             return response
-        _, system_msg = message_fit_in([{"role": "system", "content": system}], int(self._llm.max_length * 0.92))
-        response = self._llm.chat(system_msg[0]["content"], hist, conf)
-        response = re.sub(r"<think>.*</think>", "", response, flags=re.DOTALL)
-        if response.find("**ERROR**") >= 0:
-            raise Exception(response)
-        set_llm_cache(self._llm.llm_name, system, response, history, gen_conf)
-        return response
+            
+        # Implement exponential backoff retry strategy
+        max_retries = 3  # Maximum of 3 retry attempts
+        base_delay = 1.0  # Initial delay of 1 second
+        
+        for attempt in range(max_retries):
+            try:
+                # Use upstream's max_length value (0.92 instead of 0.97)
+                _, system_msg = message_fit_in([{"role": "system", "content": system}], int(self._llm.max_length * 0.92))
+                response = self._llm.chat(system_msg[0]["content"], hist, conf)
```

Comment:
`chat` at `chat_model.py:42` is a sync function. Your change will block the trio schedule thread.